src/hipoc/hipoc_program.cpp

```diff
@@ -58,30 +58,30 @@ namespace miopen {
 #if !MIOPEN_USE_COMGR
 namespace {
 
-inline bool ProduceCoV3()
+inline const std::string& GetCodeObjectVersion()
 {
-    // If env.var is set, then let's simply follow it.
-    if(IsEnabled(MIOPEN_DEBUG_OPENCL_ENFORCE_COV3{}))
-        return true;
-    if(IsDisabled(MIOPEN_DEBUG_OPENCL_ENFORCE_COV3{}))
-        return false;
-    // Otherwise, let's assume that OpenCL kernels shall be compiled to
-    // CO v3 format by default since ROCm 3.0. The simplest way to find out
-    // this right now is checking the HIP compiler version string.
-    return (HipCompilerVersion() >= external_tool_version_t{3, 0, -1});
-}
+    static const std::string opt_enable_v2{"-mno-code-object-v3"};
+    static const std::string opt_enable_v3{"-mcode-object-v3"};
+    static const std::string opt_enable_v4{"-mllvm --amdhsa-code-object-version=4"};
```

Comment:
Should the new -mcode-object-version be used?

Reply:
Totally, I could not find it last night. Do you recall new option's name?

Reply:
I believe the option is called "-mcode-object-version".

On further discussion I think the idea is to restructure this code so if the ENV variable is not present then no option is added so the compiler uses its default.

If the ENV variable is present then the option used is based on the compiler version as follows.

If compiler is <4.0 then use the -mno-code-object-v3/-mcode-object-v3 for code object V2/V3 if the ENV variable is 2/3 otherwise give an error.

If compiler is >=4.0 use -mcode-object-version=N where N is the value of the ENV variable.